### PR TITLE
feat: add kubernetes support with Helm

### PR DESCRIPTION
**Pull request by icrotz**
## What kind of change does this PR introduce?

Feature, Docs

## What is the current behavior?

For the moment the only option to deploy Supabase is using Docker Compose or Kompose.

## What is the new behavior?

Adding a Helm chart to deploy a complete supabase architecture.

By following the README.me located in the `charts` directory you can start supabase in a `minikube` cluster easily.

It's also technically possible to deploy a production environment with replication and backup, but no documentation has been written on that.

_The Supabase Studio app is not protected, don't forget to disable it before going to production. You can also choose to use a Basic HTTP authentication if you use the nginx ingress controller_

## Additional context

- You need to have a kubernetes cluster to deploy the Helm chart. You can create have locally using [Minikube](https://minikube.sigs.k8s.io/docs/start/)
- We are using the bitnami chart to deploy the postgres database. 
- We are using a custom version of the bitnami postgres image to add the extension. You might want to publish this image on your account instead.
- I created another PR to allow the modification of the url used for the API documentation because at the moment it's hard coded. [ref](https://github.com/supabase/supabase/pull/4880)
- If you try the command `helm repo add` you'll have to use ` https://koji-sas.github.io/supabase-kubernetes` url instead.
- You will have to enabled github pages on the `main` branch so Helm is able to use it as a repository

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,27 @@
...
+
+Supabase is an open source Firebase alternative. We're building the features of Firebase using enterprise-grade open source tools.
+
+## How to use ?
+
+You can find the documentation inside the [chart directory](./charts/supabase/README.md)
+
+## Support
+
+This repository is supported by the community and not officially supported. Please do not create issue on the Supabase repository if you face any error using this repository.
```
Comment (milanvanschaik) [inline README.md:17]:
```suggestion
This project is supported by the community and not officially supported by Supabase. Please do not create any issues on the official Supabase repositories if you face any problems using this repository, but rather open an issue on this repository.
```

File: README.md [hunk 2]
```diff
@@ -1 +1,27 @@
...
+
+## Contributing
+
+You can contribute to this project by forking this repository and opening a pull required
```
Comment (milanvanschaik) [inline README.md:21]:
```suggestion
You can contribute to this project by forking this repository and opening a pull request.
```

**Review by milanvanschaik — CHANGES_REQUESTED** [review]

**Comment by milanvanschaik** [comment]
Nice work (again)! I'll review it in parts over the next days.

RE Postgres on Kubernetes: maybe we can advice against it for production usage and advise using a managed Postgres instance unless people really know what they do?
In my experience it's quite tricky to configure it right, and for production environments I often choose for things like Google Cloud SQL over self-managing it. Though, this might need some changes to support an external DB aswel, and maybe it's not possible due to some exact PG configuration Supabase requires.

What are your thoughts on this?

**Comment by icrotz** [comment]
> RE Postgres on Kubernetes: maybe we can advice against it for production usage and advise using a managed Postgres instance unless people really know what they do? In my experience it's quite tricky to configure it right, and for production environments I often choose for things like Google Cloud SQL over self-managing it. Though, this might need some changes to support an external DB aswel, and maybe it's not possible due to some exact PG configuration Supabase requires.

Yes, I also prefer to use managed Postgres database. But as you said, most of the self managed database don't provide support for custom extension and are missing pgjwt required by PostRest. Also in my company we are using K8s for either production and local development. So I had to include it, that's why I included a disclamer in the chart README.

I added a `enabled` value to disable the database provisioning if needed.

**Comment by merqurio** [comment]
Hi there ! 

We use [PGO](https://access.crunchydata.com/documentation/postgres-operator/latest/) for production and we would be more than happy to extend the chart to support a production ready managed Postgres deployment. 

If this is of interest, please let us know, more than happy to put some effort on that.

**Comment by milanvanschaik** [comment]
I'm really sorry for my absence here. Busy times at our company, but I'd like to move this forward ASAP.

@icrotz You've done some excellent work! Let's move this forward.

@merqurio Didn't know about PGO, looks promising! Are you still open to help with extending the chart with PGO support? I think the best would be to base on this branch for now (or I'll give you a ping when this one is merged)

**Review by milanvanschaik — APPROVED** [review]

**Comment by merqurio** [comment]
> @merqurio Didn't know about PGO, looks promising! Are you still open to help with extending the chart with PGO support?

Sure ! I'll start working on this and create a PR

**Comment by icrotz** [comment]
Good news !

**Comment by kinrezC** [comment]
> > @merqurio Didn't know about PGO, looks promising! Are you still open to help with extending the chart with PGO support?
> 
> Sure ! I'll start working on this and create a PR

Did you ever make progress on this? I'd be open to helping out if you're looking to kickstart this initiative.

**Comment by merqurio** [comment]
yes i did! i have it on our development environment up and running, let me
push a PR this week. thanks for the reminder!

**Pull request merged**